rllib/core/rl_trainer/trainer_runner.py

```diff
@@ -128,18 +184,20 @@ def _distributed_update(self, batch: MultiAgentBatch) -> List[Mapping[str, Any]]
         """
         refs = []
         global_size = len(self._workers)
-        batch_size = math.ceil(len(batch) / global_size)
         for i, worker in enumerate(self._workers):
             batch_to_send = {}
             for pid, sub_batch in batch.policy_batches.items():
                 batch_size = math.ceil(len(sub_batch) / global_size)
                 start = batch_size * i
                 end = min(start + batch_size, len(sub_batch))
                 batch_to_send[pid] = sub_batch[int(start) : int(end)]
+            # TODO (Avnish): int(batch_size) ? How should we shard MA batches really?
             new_batch = MultiAgentBatch(batch_to_send, int(batch_size))
             refs.append(worker.update.remote(new_batch))
 
-        return ray.get(refs)
+        results = ray.get(refs)
+        # take an average across the result of all actors
+        return tree.map_structure(lambda *x: np.mean(x), *results)
```

Comment:
Dumb question: Should we be more careful here as to what the exact metric is that we average across?

Maybe not relevant for PPO, but for other algos, there is the TD-error metric, which needs to remain in-tact due to it being used as a per-single-batch-item weight in a prioritized replay buffer.

Not suggesting to provide a final solution here, but leave a TODO, if my argument above is valid.

Reply:
got it. yeah. We need a system for back-reporting metrics giving control to user on which should be averaged and which should just be concatenated.

Reply:
left a TODO